src/init/variables/VarResolver.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable tsdoc/syntax */
+import yargs from 'yargs';
+import { createErrorMessage } from '../..';
+import { AsyncHandler } from '../../util/handlers/AsyncHandler';
+import { modulePathPlaceholder } from '../../util/PathUtil';
+import type { VarComputer } from './VarComputer';
+
+const defaultConfig = `${modulePathPlaceholder}config/default.json`;
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+export type YargsArgOptions = Record<string, yargs.Options>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function setupYargvWithBaseArgs(yargv: yargs.Argv) {
+  // Values from componentsjs LogLevel type
+  const logLevels = [ 'error', 'warn', 'info', 'verbose', 'debug', 'silly' ];
+  return yargv.options({
+    /**
+   * CLI options needed for performing meta-process of app initialization.
+   * These options doesn't contribute to components-js vars normally.
+   */
+    config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+    loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true, choices: logLevels },
+    mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
+    varConfig: { type: 'string', alias: 'v', default: defaultVarConfig, requiresArg: true },
+  });
+}
+
+export interface YargvOptions {
+  usage?: string;
+  strictMode?: boolean;
+  // @see http://yargs.js.org/docs/#api-reference-envprefix
+  loadFromEnv?: boolean;
+  envVarPrefix?: string;
+}
+
+export type VarRecord = Record<string, unknown>;
+
+export class VarResolver extends AsyncHandler<string[], VarRecord> {
```

Comment:
```suggestion
/**
 * This class translates command-line arguments into values for specific variables,
 * which can then be used to instantiate a parametrized Components.js configuration.
 */
export class CliVariableResolver extends AsyncHandler<string[], VarRecord> {
```

Reply:
it resolves values of `componentsjs variables` from both `cli` and `env` as per configuration.

Reply:
`CliVariableResolver` is fine; `env` is still a command-line interface construct.